intezer_sdk/analysis.py

```diff
@@ -66,16 +69,21 @@ def send(self, wait: typing.Union[bool, int] = False) -> None:
 
         if wait:
             if isinstance(wait, int):
-                self.wait_for_completion(wait, sleep_before_first_check=True)
+                self.wait_for_completion(wait, sleep_before_first_check=True, timeout=wait_timeout)
             else:
-                self.wait_for_completion(sleep_before_first_check=True)
+                self.wait_for_completion(sleep_before_first_check=True, timeout=wait_timeout)
 
-    def wait_for_completion(self, interval: int = None, sleep_before_first_check=False):
+    def wait_for_completion(self,
+                            interval: int = None,
+                            sleep_before_first_check=False,
+                            timeout: typing.Optional[datetime.timedelta] = None):
         """
         Blocks until the analysis is completed
         :param interval: The interval to wait between checks
         :param sleep_before_first_check: Whether to sleep before the first status check 
+        :param timeout: Maximum duration to wait for analysis completion
         """
+        start_time = datetime.datetime.now()
```

Comment:
as a best practice - always use `datetime.utcnow()` and not `datetime.now()`

Reply:
done and done